### PR TITLE
Strict types for `chrome.{tabs,extension}.sendRequest`

### PR DESCRIPTION
**Pull request by bershanskiy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developer.chrome.com/docs/extensions/reference/extension/#method-sendRequest>
<https://developer.chrome.com/docs/extensions/reference/tabs/#method-sendRequest>

This PR adds tests and introduces type arguments with default values any. This change is backwards-compatiable because existing code will just use default values of any which will match prior behavior.

### CODE REVIEW

**Comment by nicolas377** [comment]
The .sendRequest family has been deprecated in favour of .sendMessage

**Comment by bershanskiy** [comment]
> The .sendRequest family has been deprecated in favour of .sendMessage

Does that mean that type declarations are not being updated? I'm not using `sendRequest` in any of my code, I just decided to update declarations.

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -3316,15 +3316,15 @@ declare namespace chrome.extension {
      * function(any response) {...};
      * Parameter response: The JSON response object sent by the handler of the request. If an error occurs while connecting to the extension, the callback will be called with no arguments and runtime.lastError will be set to the error message.
      */
-    export function sendRequest(extensionId: string, request: any, responseCallback?: (response: any) => void): void;
+    export function sendRequest<T = any, R = any>(extensionId: string, request: T, responseCallback?: (response: R) => void): void;
```
Comment (nicolas377) [inline types/chrome/index.d.ts:3319]: could you make the variables more readable?

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -3316,15 +3316,15 @@ declare namespace chrome.extension {
...
     /**
      * Sends a single request to other listeners within the extension. Similar to runtime.connect, but only sends a single request with an optional response. The extension.onRequest event is fired in each page of the extension.
      * @deprecated Deprecated since Chrome 33. Please use runtime.sendMessage.
      * @param responseCallback If you specify the responseCallback parameter, it should be a function that looks like this:
      * function(any response) {...};
      * Parameter response: The JSON response object sent by the handler of the request. If an error occurs while connecting to the extension, the callback will be called with no arguments and runtime.lastError will be set to the error message.
      */
-    export function sendRequest(request: any, responseCallback?: (response: any) => void): void;
+    export function sendRequest<T = any, R = any>(request: T, responseCallback?: (response: R) => void): void;
```
Comment (nicolas377) [inline types/chrome/index.d.ts:3327]: same thing over here

File: types/chrome/index.d.ts [hunk 3]
```diff
@@ -8983,7 +8983,7 @@ declare namespace chrome.tabs {
      * @param responseCallback Optional.
      * Parameter response: The JSON response object sent by the handler of the request. If an error occurs while connecting to the specified tab, the callback will be called with no arguments and runtime.lastError will be set to the error message.
      */
-    export function sendRequest(tabId: number, request: any, responseCallback?: (response: any) => void): void;
+    export function sendRequest<T = any, R = any>(tabId: number, request: T, responseCallback?: (response: R) => void): void;
```
Comment (nicolas377) [inline types/chrome/index.d.ts:8986]: and here

**Review by nicolas377 — CHANGES_REQUESTED** [review]
just make T and R more readable

File: types/chrome/index.d.ts (hunk 1 above)
Comment (bershanskiy) [reply types/chrome/index.d.ts]: I changed them to `Request` and `Response`. If you have better names in mind, I'd gladly use that.

Comment (nicolas377) [reply types/chrome/index.d.ts]: they look good to me! go ahead and mark these conversations as resolved

**Review by nicolas377 — APPROVED** [review]
LGTM!

**Comment by nicolas377** [comment]
Ready to merge

**Pull request merged**